src/routes/index.svelte

```diff
@@ -55,28 +47,127 @@
 
 <script>
 	import { onMount } from 'svelte';
+	import { Switch } from 'spaper';
 	import Map from '$lib/components/Map.svelte';
-
+	import { MetricEnum } from '../types';
+	export let metric = MetricEnum.NONE;
+	let metricsAvailable = false;
+	let switch1 = false;
+	let switch2 = false;
+	let switch3 = false;
+	const updateMetric = (m, v) => {
+		// TODO can this easily be generalized?
+		console.log('here');
+		switch (m) {
+			case MetricEnum.VULNERABLE_ROAD_USER_TO_CAR: {
+				if (switch1 === false) {
+					if (switch2 || switch3) {
+						// This is only triggered by enabling another metric.
+						// Ignore in this case.
+						return;
+					}
+					metric = MetricEnum.NONE;
+				} else {
+					metric = m;
+					switch2 = false;
+					switch3 = false;
+				}
+				break;
+			}
+			case MetricEnum.TOO_FAST_120_PERCENT: {
+				if (switch2 === false) {
+					if (switch1 || switch3) {
+						// This is only triggered by enabling another metric.
+						// Ignore in this case.
+						return;
+					}
+					metric = MetricEnum.NONE;
+				} else {
+					metric = m;
+					switch1 = false;
+					switch3 = false;
+				}
+				break;
+			}
+			case MetricEnum.TOO_FAST_120_PERCENT_PLUS: {
+				if (switch3 === false) {
+					if (switch1 || switch2) {
+						// This is only triggered by enabling another metric.
+						// Ignore in this case.
+						return;
+					}
+					metric = MetricEnum.NONE;
+				} else {
+					metric = m;
+					switch1 = false;
+					switch2 = false;
+				}
+				break;
+			}
+		}
+	};
+	$: switch1 && updateMetric(MetricEnum.VULNERABLE_ROAD_USER_TO_CAR, true);
+	$: !switch1 && updateMetric(MetricEnum.VULNERABLE_ROAD_USER_TO_CAR, false);
+	$: switch2 && updateMetric(MetricEnum.TOO_FAST_120_PERCENT, true);
+	$: !switch2 && updateMetric(MetricEnum.TOO_FAST_120_PERCENT, false);
+	$: switch3 && updateMetric(MetricEnum.TOO_FAST_120_PERCENT_PLUS, true);
+	$: !switch3 && updateMetric(MetricEnum.TOO_FAST_120_PERCENT_PLUS, false);
 	export let snapshot = [];
-	//let speedData = [];
-	// onMount(
-	//     Promise.all(
-	//         snapshot.features.map(segment => {
-	//             return fetch(`/api/speed-${segment.properties.segment_id}`)
-	//         })
-	//     ).then(function (responses) {
-	//         return Promise.all(responses.map(res => {
-	//             return res.json()
-	//         }));
-	//     }).then(speed => {
-	//         console.log(speed.report);
-	//         speedData = speed.report;
-	//     }).catch(error => {
-	//         console.log(error);
-	//     })
-	// );
-	//$: segments = snapshot.features.map(x => x.properties.segment_id)
+	const timeEnd = new Date();
+	let timeStart = new Date(timeEnd);
+	timeStart.setMonth(timeStart.getMonth() - 1);
```

Comment:
Is the monthly counts the one we are most interested in? Or would 'last week' make more sense?

Reply:
Updated, it will also make the website a bit more responsive.

Reply:
Updated, it will also make the website a bit more responsive.